src/SonarScanner.MSBuild.PreProcessor/PreprocessorObjectFactory.cs

```diff
@@ -155,5 +161,25 @@ static async Task<Version> GetVersion(IDownloader downloader, string path)
                 return new Version(contents.Split('-')[0]);
             }
         }
+
+        /// <summary>
+        /// Makes a throwaway request to the server to ensure we can properly authenticate.
+        /// This might fail in the scenario where the user does not specify sonar.host.url.
+        /// Before 7.0: Defaulted to "http://localhost:9000".
+        /// After 7.0: Defaults to "https://sonarcloud.io".
+        /// </summary>
+        private async Task<bool> EnsureServerIsSonarCloud(SonarCloudWebServer server)
```

Comment:
I would rename the method to `CanAuthenticate` and return false if the HTTPCode is 403 or 401. 

Regarding the logging message, I think we should begin by telling that the authentication parameters provided are not valid and that the authentication fails. The continue with "Did you want to connect to a local SonarQube server?" ...

Reply:
We discussed offline and a possible solution is to call
```
var httpResponseMessage = await server.DownloadResource(WebUtils.Escape("api/settings/values?component=unknown"));
```
and read the HTTP status code of the response message.